bubble_sorting.rb

```diff
@@ -0,0 +1,28 @@
+def bubble_sort(arr)
+  arr.each_with_index do |_value, _i|
+    arr.each_with_index do |_val, index|
+      break if index + 1 == arr.length
+
+      arr[index] > arr[index.next] && (arr[index], arr[index.next] = arr[index.next], arr[index])
+    end
+    arr
+  end
+end
+
+puts bubble_sort([3, 0, 128 - 3, -65, 45, 4, 5, 1, 10, 2])
+
+def bubble_sort_by(arr)
+  arr.each_with_index do |_value, _i|
+    arr.each_with_index do |_val, index|
+      break if index.next == arr.length
+
+      difference = yield(arr[index], arr[index.next])
+      difference.positive? && (arr[index], arr[index.next] = arr[index.next], arr[index])
+    end
+  end
+  puts arr
+end
```

Comment:
## Debugging code 
- No debugging instructions in committed code (as discussed [here](https://questions.microverse.org/t/debugging-instructions-code/346)) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/section/debugging_requirement.png) from Pathwright)
  - [ ] Please remove a debugging code `p`, and return only the arr var